packages/core-modules/contracts/modules/CoreElectionModule.sol

```diff
@@ -37,4 +43,58 @@ contract CoreElectionModule is IElectionModule, ElectionStorage, OwnableMixin {
     function getMemberTokenAddress() public view override returns (address) {
         return _electionStore().memberTokenAddress;
     }
+
+    function getNominees() external view override returns (address[] memory) {
+        return _electionStore().nominees;
+    }
+
+    function selfNominate() external override {
+        ElectionStore storage store = _electionStore();
+
+        if (msg.sender == address(0)) {
+            revert AddressError.ZeroAddress();
+        }
+
+        if (store.nomineesIndexes[msg.sender] != 0) {
+            revert AlreadyNominated(msg.sender);
+        }
+
+        store.nominees.push(msg.sender);
+        store.nomineesIndexes[msg.sender] = store.nominees.length;
+    }
+
+    function selfUnnominate() external override {
+        ElectionStore storage store = _electionStore();
+
+        uint256 valueIndex = store.nomineesIndexes[msg.sender];
```

Comment:
Do we really need information about the index of the nominee?
Can't we use a boolean dictionary to just know if the nominee is in the array? That way all this removal logic becomes unnecessary.

Reply:
Hmm, not sure how would that work without having to potentially iterate the entire array on removals, which could be a gas limited operation. Also, we need to keep the array up to date so we can return it on `getNominees()`.

Reply:
Shit yeah. I brainfarted. Please ignore!